### DISCUSSION TITLE
How to properly re-join / heal devices?

### DISCUSSION THREAD

**Discussion by redm123** [Koenkk/zigbee2mqtt#24665] [General]
Hi

I have a few Aqara devices (mostly door/window sensors, environment sensors). Unfortunately they get kinda stuck from time to time and don't deliver any more data. The only way I found to fix this is force removing the device from Z2M and rejoining it.
Problem with that is, that I loose all the device config (name, description, other settings). It's cumbersome to put that all back together. And esp. if you have multiple of the same kind it easy to mix things up...

Is there a better way to do this? It would already be helpful if Z2M would persist the device config and pick it up again when the device re-joins. Or automatically to remove and re-join a device? Didn't find anything in that direction, though..

**Comment by andreasbrett** [comment]
For me it works by simply pairing again (enable "permit join" and initiate pairing based on zigbee device). I do NOT delete the device from zigbee2mqtt and thus only the connection is restored with all settings intact.

**Comment by redm123** [comment]
Hmm, not sure I tried that...🤔 I tried various things, before removing. But not sure if it was trigger join AND permit join AND really being close enough to the controller... But if you say that works for you, I'll try specifically that next time it happens. Thanks for the hint!

**Comment by hasherdk** [comment]
I have the same issue. Most devices are rock stable, but just a few of them very frequently (like weekly or more) need re-pairing. For me it also works to just pair it without first removing, but it's still something that must be done manually, when I find out that data is not updating.

Do you have any idea how to fix this, besides replacing the unstable devices?

**Reply by andrei-lazarov** [reply]
Which models are those?

**Reply by redm123** [reply]
Yes, for me it also works with just repairing. Although it can sometime take a few tries, or maybe some time to detect proximity to the controller, or take the battery out, no idea, for join to work. But in the end it always worked, and I didn't have to remove and re-add them.

More than weekly is certainly annoying. But not sure if this is specific to certain devices. I mean, I don't have to do it weekly normally. Every few months one refuses to answer, some more likely than others. Like the Aqara environment sensors more likely than the Aqara contact sensors, and the Aqara buttons never up to now. But it could also be their location or whatever... I didn't try to move them elsewhere to see if this changes anything. In my case for each one them this just happens to infrequently to collect some meaningful statistics... Did you maybe try that, for one of the frequent candidates?

**Reply by andrei-lazarov** [reply]
Those connection issues are indeed specific to Aqara battery-powered devices. I don't think shifting them around will help.
Update their firmware if there is any available (look up the models in Koenkk/zigbee-ota), or replace the unstable devices with other brands.

**Reply by redm123** [reply]
Specific to Aqara devices, yes. I meant specific to a particular device, and whether it makes sense to replace by one of the same type. Or if there are other circumstances that trigger that behavior in Aqara devices... but yea, just guessing.

> replace the unstable devices with other brands.

Unfortunately the Aqara devices are pretty neat: Good looking, small, no exotic expensive battery, long lasting, generally working well, comparatively cheap. Haven't seen others the like. But yea, guess there is also a downside... But with some alerts around, at least for me still ok.

> Update their firmware if there is any available

Does not look like...

For the record, these are my device models which are a bit capricious. All devices with the same firmware version, as far as I can tell.

Zigbee Model: lumi.weather
Zigbee Manufacturer: LUMI
Description: Temperature and humidity sensor
Firmware build date: 20191205
Firmware version: 3000-0001
Manufacturer: Aqara
Model: WSDCGQ11LM

Zigbee Model: lumi.sensor_magnet.aq2
Zigbee Manufacturer: LUMI
Description: Door and window sensor
Firmware build date: 20161128
Firmware version: 3000-0001
Manufacturer: Aqara
Model: MCCGQ11LM
